### PR TITLE
Add Kafka Producer and Kafka Consumer as Execution Listener

### PR DESCRIPTION
**Pull request by aurelio-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE
https://aottech.atlassian.net/browse/FWF-2188

# Changes

Added a Kafka Producer as Execution Listener.  The Kafka Producer configurations are being passed as field injections.  Example of configurations are bootstrap_servers_config, key_serializer_class_config, etc.  Complete list of configurations can be found in https://kafka.apache.org/34/javadoc/org/apache/kafka/clients/producer/ProducerConfig.html.

# How has the change been tested?

The change is tested using manual verification.

image

# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

image

# Notes

### CODE REVIEW

**Review by SnehaSuresh-aot — CHANGES_REQUESTED** [review]
Its better to use `org.apache.commons.lang3.StringUtils`  instead of adding a new dependency

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/execution/KafkaProducerListener.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
+package org.camunda.bpm.extension.hooks.listeners.execution;
+
+import org.apache.kafka.clients.producer.*;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.ExecutionListener;
+import java.util.Properties;
+
+import org.apache.commons.lang.StringUtils;
```
[the remark this answers is not in the document]
Comment (aurelio-aot) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/execution/KafkaProducerListener.java]: @SnehaSuresh-aot - new commit is now pushed to address this comment

**Review by auslin-aot — COMMENTED** [review]
Just asking with curiosity the kafka consumer also will be added as a listener right?

**Comment by aurelio-aot** [comment]
> Just asking with curiosity the kafka consumer also will be added as a listener right?

i honestly have not figured/decided that out yet, because a kafka consumer is a listener on a kafka topic as opposed to being a listener of a camunda process task, feel free to let me know what you think the kafka consumer should be, any suggestions/help would be highly appreciated, thanks.

**Comment by auslin-aot** [comment]
> > Just asking with curiosity the kafka consumer also will be added as a listener right?
> 
> i honestly have not figured/decided that out yet, because a kafka consumer is a listener on a kafka topic as opposed to being a listener of a camunda process task, feel free to let me know what you think the kafka consumer should be, any suggestions/help would be highly appreciated, thanks.

Using the consumer as a  Listener on Kafka Topic will be good. Maybe we can capture future discussion points in the ticket https://aottech.atlassian.net/browse/FWF-2188.

**Review by auslin-aot — APPROVED** [review]

**Comment by aurelio-aot** [comment]
Hi @abhilash-aot @SnehaSuresh-aot @auslin-aot -- added codes and documentation for the KafkaConsumerListener, kindly review.  Thanks.

**Review by SnehaSuresh-aot — APPROVED** [review]

**Review by abhilash-aot — APPROVED** [review]

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**